api-gateway/src/main/java/com/petclinic/bffapigateway/domainclientlayer/AuthServiceClient.java

```diff
@@ -143,39 +148,52 @@ public Mono<OwnerResponseDTO> createUserUsingV2Endpoint(Mono<Register> model) {
-                                            x -> new GenericHttpException(x.get("message").toString(), BAD_REQUEST))
-                            )
-                            .bodyToMono(UserPasswordLessDTO.class)
-                            .flatMap(userDetails -> {
-                                Mono<OwnerRequestDTO> ownerRequestDTO = Mono.just(OwnerRequestDTO.builder()
-                                        .firstName(register.getOwner().getFirstName())
-                                        .lastName(register.getOwner().getLastName())
-                                        .address(register.getOwner().getAddress())
-                                        .city(register.getOwner().getCity())
-                                        .province(register.getOwner().getProvince())
-                                        .telephone(register.getOwner().getTelephone())
-                                        .ownerId(uuid)
-                                        .build());
-
-                                return customersServiceClient.addOwner(ownerRequestDTO);
-                                    }
-                            );
-                }
-        ).doOnError(throwable -> {
+            register.setUserId(uuid);
+            return webClientBuilder.build().post()
+                    .uri(authServiceUrl + "/users")
+                    .body(Mono.just(register), Register.class)
+                    .accept(MediaType.APPLICATION_JSON)
+                    .retrieve()
+                    .onStatus(HttpStatusCode::is4xxClientError,
+                            n -> rethrower.rethrow(n,
+                                    x -> new GenericHttpException(x.get("message").toString(), BAD_REQUEST))
+                    )
+                    .bodyToMono(UserPasswordLessDTO.class)
+                    .flatMap(userDetails -> {
+                        Mono<OwnerRequestDTO> ownerRequestDTO = Mono.just(OwnerRequestDTO.builder()
+                                .firstName(register.getOwner().getFirstName())
+                                .lastName(register.getOwner().getLastName())
+                                .address(register.getOwner().getAddress())
+                                .city(register.getOwner().getCity())
+                                .province(register.getOwner().getProvince())
+                                .telephone(register.getOwner().getTelephone())
+                                .ownerId(uuid)
+                                .build());
+
+                        return customersServiceClient.addOwner(ownerRequestDTO)
+                                .flatMap(ownerResponse -> {
+                                    String customerId = ownerResponse.getOwnerId();
+
+                                    /* intellij says that this is never used but for some unknown dark magic
+                                    retarded reason it breaks everything if u remove it DO NOT!! - alex xoxoxo mwah */
```

Comment:
This should be investigated further.. Since you should be able to remove it without causing any problems. If it bugs while removing it, it might be a bigger problem